app/server/utils/ussd/kenya_ussd_processor.py

```diff
@@ -71,10 +71,31 @@ def replace_vars(menu: UssdMenu, ussd_session: UssdSession, display_text: str, u
                 # TODO(ussd): replace this with i18n placeholders
                 if user.preferred_language == 'sw_KE':
                     # TODO(ussd): pin_failed_attempts isn't a thing yet!
-                    replacements.append(['%remaining_attempts%', "Una majaribio {} yaliyobaki.".format(3 - user.pin_failed_attempts)])
+                    replacements.append(
+                        ['%remaining_attempts%', "Una majaribio {} yaliyobaki.".format(3 - user.pin_failed_attempts)])
                 else:
-                    replacements.append(['%remaining_attempts%', "You have {} attempts remaining.".format(3 - user.pin_failed_attempts)])
+                    replacements.append(['%remaining_attempts%',
+                                         "You have {} attempts remaining.".format(3 - user.pin_failed_attempts)])
             else:
                 replacements.append(['%remaining_attempts%', ''])
 
         return reduce(lambda text, r: text.replace(r[0], r[1]), replacements, display_text)
+
+    @staticmethod
+    def authorize_pin(user: User, pin):
```

Comment:
is this used in the state machine? why is it here and not in the state machine?

Reply:
or it might make more sense in User if we plan on using it in places beyond just the state machine

Reply:
Might need to find out why it was put there in the first place but yes your suggestion does make sense. In the meantime let me work on the pin storage as salted and refactor a couple of the things you've brought up so that we can close on the pin auth feature.

Reply:
sounds good, you can leave the auth / actual usage of the pin for the next ticket. as part of this one just do setting pin and seeing that it works in the state machine, and being able to change the pin and seeing that works in the state machine.